manim/mobject/opengl/opengl_mobject.py

```diff
@@ -103,6 +103,64 @@ class MobjectStatus:
     points_changed: bool = False
 
 
+_O = TypeVar("_O")
+
+
+def affects_color(func) -> Callable[..., _O]:
```

Comment:
```suggestion
_F = TypeVar("_F", bound=Callable[..., Any])
def affects_color(func: _F) -> _F:
```
See <https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators>

Reply:
@JasonGrace2282 As mentioned a few lines below, I tend to favor the new `ParamSpec` type and usage, e.g.,:

```python
from typing import Callable, TypeVar
from typing_extensions import ParamSpec

P = ParamSpec('P')
T = TypeVar('T')

 # We reuse 'P' in the return type, but replace 'T' with 'str'
def stringify(func: Callable[P, T]) -> Callable[P, str]:
    def wrapper(*args: P.args, **kwds: P.kwargs) -> str:
        return str(func(*args, **kwds))
    return wrapper

 @stringify
 def add_forty_two(value: int) -> int:
     return value + 42
```

Reply:
Interesting, didn't realize `ParamSpec` had been backported to `typing_extensions`!